### PR TITLE
Editor: Refactor `PostPublishPanel` tests to `@testing-library/react`

### PR DESCRIPTION
**Pull request by tyxla**
## What?
We've recently started refactoring `enzyme` tests to `@testing-library/react`.

This PR refactors the `<PostPublishPanel />` component tests from `enzyme` to `@testing-library/react`.

## Why?
`@testing-library/react` provides a better way to write tests for accessible components that is closer to the way the user experiences them.

Ideally, we should not be using snapshot tests here, however, the purpose of this PR is not to improve or enrich the tests themselves, but rather to migrate them `@testing-library/react`. Our primary motivation is unblocking the upgrade to React 18. 

## How?
We're straightforwardly replacing `enzyme` tests with `@testing-library/react` ones, using `jest-dom` matchers and mocks to avoid testing unrelated implementation details.

## Testing Instructions
Verify tests pass: `npm run test:unit packages/editor/src/components/post-publish-panel/test/index.js`

### CODE REVIEW

File: packages/editor/src/components/post-publish-panel/test/__snapshots__/index.js.snap [hunk 1]
```diff
@@ -1,197 +1,733 @@
...
+              class="components-base-control post-publish-panel__postpublish-post-address emotion-0 emotion-1"
+            >
+              <div
+                class="components-base-control__field emotion-2 emotion-3"
+              >
+                <label
+                  class="components-base-control__label emotion-4 emotion-5"
+                  for="inspector-text-control-0"
+                >
+                  undefined address
```
Comment (ciampo) [inline packages/editor/src/components/post-publish-panel/test/__snapshots__/index.js.snap]: This `undefined` is interesting, may be worth looking into that a bit more to see if it's expected

File: packages/editor/src/components/post-publish-panel/test/__snapshots__/index.js.snap [hunk 2]
```diff
@@ -1,197 +1,733 @@
...
+                </label>
+                <input
+                  class="components-text-control__input"
+                  id="inspector-text-control-0"
+                  readonly=""
+                  type="text"
+                  value="undefined"
```
Comment (ciampo) [inline packages/editor/src/components/post-publish-panel/test/__snapshots__/index.js.snap]: Another `undefined` 🤔  I'm not sure this is valid HTML at all ?

File: packages/editor/src/components/post-publish-panel/test/__snapshots__/index.js.snap [hunk 3]
```diff
@@ -1,197 +1,733 @@
...
+            class="post-publish-panel__postpublish-buttons"
+          >
+            <button
+              class="components-button is-primary"
+              type="button"
+            />
+            <a
+              class="components-button is-secondary"
+              href="post-new.php?"
+            />
```
Comment (ciampo) [inline packages/editor/src/components/post-publish-panel/test/__snapshots__/index.js.snap]: These two elements have a missing label. Looking more into this, it's related to the previous "undefined address" comment — the "missing" values all come from a ` get( postType, [ 'labels', xxx ] );` type call

**Review by ciampo — APPROVED** [review]
The refactor in itself looks good! 🚀  


although now that the snapshots are showing the actual rendered DOM, there may be a few issues that we'd like to address (see my inline comments). If we were to write these tests with more details assertions (i.e look for a button with a certain label, look for the post title's text etc), those assertions would currently fail. I suspect it's related to the need of mocking the data layer.

I understand that this PR focuses on refactoring to RTL, so feel free to merge it — but we should definitely make a note and iterated on these tests, because they currently offer very little value

**Comment by tyxla** [comment]
Thanks for taking a look, @ciampo! Seems like all these were from missing mocks in the original tests. In 9b8da201a0c63fb1fc7be29031bf0215277b8731 I've added a few mocks and the snapshots no longer have undefined values.

File: packages/editor/src/components/post-publish-panel/test/__snapshots__/index.js.snap (hunk 1 above)
Comment (tyxla) [reply packages/editor/src/components/post-publish-panel/test/__snapshots__/index.js.snap]: Addressed as part of https://github.com/WordPress/gutenberg/commit/9b8da201a0c63fb1fc7be29031bf0215277b8731.

File: packages/editor/src/components/post-publish-panel/test/__snapshots__/index.js.snap (hunk 2 above)
Comment (tyxla) [reply packages/editor/src/components/post-publish-panel/test/__snapshots__/index.js.snap]: Addressed as part of https://github.com/WordPress/gutenberg/commit/9b8da201a0c63fb1fc7be29031bf0215277b8731.

File: packages/editor/src/components/post-publish-panel/test/__snapshots__/index.js.snap (hunk 3 above)
Comment (tyxla) [reply packages/editor/src/components/post-publish-panel/test/__snapshots__/index.js.snap]: Addressed as part of https://github.com/WordPress/gutenberg/commit/9b8da201a0c63fb1fc7be29031bf0215277b8731.

**Pull request merged**